cmake/AwsCRuntime.cmake

```diff
@@ -0,0 +1,29 @@
+# Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+# SPDX-License-Identifier: Apache-2.0.
+
+# This function detects the local host's c runtime and writes a tag into the supplied output variable
+# Output is "cruntime" on non-Linux platforms.  Output is "glibc" or "musl" on Linux platforms.
+#
+# Intended usage is for managed language CRTs to use this function to build native artifact paths, facilitating
+# support for alternative C runtimes like Musl.
+#
+function(aws_determine_local_c_runtime target)
+    if (CMAKE_SYSTEM_NAME STREQUAL "Linux")
+        execute_process(COMMAND "ldd" "--version" OUTPUT_VARIABLE AWS_LDD_OUTPUT ERROR_VARIABLE AWS_LDD_OUTPUT)
+        string(FIND "${AWS_LDD_OUTPUT}" "musl" AWS_MUSL_INDEX)
+        string(FIND "${AWS_LDD_OUTPUT}" "GLIBC" AWS_GLIBC_INDEX)
```

Comment:
just FYI the output on my dev-desktop doesn't have "GLIBC", so it's going to print "Could not determine C runtime"
> ldd (GNU libc) 2.26
> Copyright (C) 2017 Free Software Foundation, Inc.
> This is free software; see the source for copying conditions.  There is NO
> warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
> Written by Roland McGrath and Ulrich Drepper.

Not sure if it's smarter to: try more pattern matches, or don't even bother with explicit checks for "glibc" 🤷‍♀️

Reply:
Will add GNU I guess